### ISSUE TITLE
Is there a  way to populate <input type="month" using <DateInput/>

### ISSUE THREAD

**Issue by pushpendra2011** [marmelab/react-admin#6487]
For one of the scenarios, we want to populate a field with just month and year fields.`<DateInput/>` by default renders 
`<input type="date"/>`. I tried replacing the type attribute of `DateInput` and it did help in rendering the month and years however, I was not able to capture the values being selected.
![image](https://user-images.githubusercontent.com/58908788/128152283-0b9f2441-7697-4d7e-8212-cef6f92d4556.png)

**Comment by djhi** [comment]
Thanks for reporting this. Please provide a sample application showing the issue by forking the following CodeSandbox (https://codesandbox.io/s/github/marmelab/react-admin/tree/master/examples/simple).

**Comment by Pushpendra92** [comment]
@djhi  -- For some reason the link you shared above giving me error (Error starting the sandbox). I am sharing my earlier forked code sandbox here if it helps.
https://codesandbox.io/s/react-admin-demo-300-forked-7rk3j?file=/src/orders/OrderEdit.js

I have made changes in `OrderEdit.js`, in useEffect I am changing the type of the input. 
```js
<DateInput
          id="monthCal"
          format={dateFormatter}
          parse={dateParser}
          // type="month"
          // options={{ format: "MM/YYYY", clearable: true }}
          source="date"
        />
```

I had tried something with `dateFormatter` and `dateParser` but with no success, the behavior was same. I am adding the video snippet


https://user-images.githubusercontent.com/58908788/128224266-f9fb1b7a-1d46-4c8c-bf29-6edb977414f8.mp4
